server/src/test/java/org/elasticsearch/cluster/routing/allocation/ShardsHealthIndicatorServiceTests.java

```diff
@@ -0,0 +1,310 @@
+                        List.of(),
+                        List.of(restartingIndex),
+                        List.of()
+                    )
+                )
+            )
+        );
+    }
+
+    private HealthIndicatorResult createExpectedResult(HealthStatus status, String summary, SimpleHealthIndicatorDetails details) {
+        return new HealthIndicatorResult(NAME, DATA, status, summary, details);
+    }
+
+    private SimpleHealthIndicatorDetails createDetails(
+        List<IndexRoutingTable> allocatedPrimaries,
+        List<IndexRoutingTable> allocatedReplicas,
+        List<IndexRoutingTable> unreplicatedPrimaries,
+        List<IndexRoutingTable> unallocatedReplicas,
+        List<IndexRoutingTable> restartingReplicas,
+        List<IndexRoutingTable> unallocatedPrimaries
+    ) {
+        return new SimpleHealthIndicatorDetails(
+            Map.of(
+                "allocated_primaries_count",
+                allocatedPrimaries.size(),
+                "allocated_replicas_count",
+                allocatedReplicas.size(),
+                "unreplicated_primaries_count",
+                unreplicatedPrimaries.size(),
+                "unreplicated_primaries",
+                unreplicatedPrimaries.stream().map(it -> it.shards().get(1).shardId()).toList(),
+                "unallocated_replicas_count",
+                unallocatedReplicas.size(),
+                "unallocated_replicas",
+                unallocatedReplicas.stream().map(it -> it.shards().get(1).shardId()).toList(),
+                "restarting_replicas_count",
+                restartingReplicas.size(),
+                "restarting_replicas",
+                restartingReplicas.stream().map(it -> it.shards().get(1).shardId()).toList(),
+                "unallocated_primaries_count",
+                unallocatedPrimaries.size(),
+                "unallocated_primaries",
+                unallocatedPrimaries.stream().map(it -> it.shards().get(1).shardId()).toList()
+            )
+        );
+    }
+
+    private static ClusterState createClusterStateWith(List<IndexRoutingTable> indexes) {
+        var builder = RoutingTable.builder();
+        for (IndexRoutingTable index : indexes) {
+            builder.add(index);
+        }
+        return ClusterState.builder(new ClusterName("test-cluster"))
+            .routingTable(builder.build())
+            .metadata(Metadata.builder().putCustom(NodesShutdownMetadata.TYPE, new NodesShutdownMetadata(Map.of())).build())
+            .build();
+    }
+
+    private static Supplier<IndexRoutingTable> indexGenerator(String prefix, ShardState primaryState, ShardState... replicaStates) {
+        var index = new AtomicInteger(0);
+        return () -> index(prefix + index.incrementAndGet(), primaryState, replicaStates);
+    }
+
+    private static IndexRoutingTable index(String name, ShardState primaryState, ShardState... replicaStates) {
+        var index = new Index(name, UUID.randomUUID().toString());
```

Comment:
Nit: If you want to use an UUID here, I believe it needs to Base64 encoded, so it's better to use `UUIDs.randomBase64UUID()`, but I also think that any random string would be sufficient.

Reply:
`UUIDs.randomBase64UUID(random())` allows to have reproducible UUIDs which is handy in randomized tests.